workspaces/src/types/gas_meter.rs

```diff
@@ -0,0 +1,76 @@
+use std::{
+    ops::AddAssign,
+    sync::{Arc, Mutex},
+};
+
+use crate::Worker;
+
+use super::Gas;
+use crate::result::Result;
+
+/// A hook that is called on every transaction that is sent to the network.
+/// This is useful for debugging purposes, or for tracking the amount of gas
+/// that is being used.
+pub type GasHook = Arc<Mutex<dyn FnMut(Gas) -> Result<()> + Send>>;
```

Comment:
we actually don't need to have a mutex here. The function never gets mutated, so this isn't necessary anymore.
```suggestion
pub type GasHook = Arc<dyn Fn(Gas) -> Result<()> + Send + Sync>;
```